api/src/test/java/org/apache/iceberg/AssertHelpers.java

```diff
@@ -68,17 +68,43 @@ public static void assertThrows(String message,
     }
   }
 
+  /**
+   * A convenience method to avoid a large number of @Test(expected=...) tests
+   * @param message A String message to describe this assertion
+   * @param expected An Exception class that the Runnable should throw
+   * @param callable A Callable that is expected to throw the exception
+   */
+  public static void assertThrows(String message,
```

Comment:
Seems like we could have avoid these methods if we throw NoSuchElementException with a message. But I guess I'm OK with these as well.